pkg/processor/eventprocessor/pool.go

```diff
@@ -32,51 +31,42 @@ type syncPoolAllocator struct {
 	logger      logger.Logger
 	objectsChan chan EventProcessor
 	objects     []EventProcessor
-	statistics  AllocatorStatistics
+	statistics  safeAllocatorStatistics
 
 	isTerminated bool
 }
 
 func NewSyncPoolAllocator(parentLogger logger.Logger, objects []EventProcessor) Allocator {
 	newFixedPool := &syncPoolAllocator{
-		logger:      parentLogger.GetChild("sync_pool_allocator"),
-		objectsChan: make(chan EventProcessor, len(objects)),
-		objects:     objects,
-		statistics:  AllocatorStatistics{},
+		logger:     parentLogger.GetChild("sync_pool_allocator"),
+		statistics: safeAllocatorStatistics{},
 	}
-
-	// iterate over objects, shove to pool
-	for _, object := range objects {
-		newFixedPool.objectsChan <- object
+	if err := newFixedPool.SetObjects(objects); err != nil {
+		return nil
```

Comment:
Why does return nil on err check is legit?

Reply:
@liranbg `newFixedPool.SetObjects(objects)` doesn't return any error

Reply:
maybe now it doesn't, but if the interface does return `error` than it's good practice to always check it, and at the very least log it.